modules/imaging_browser/php/NDB_Menu_Filter_imaging_browser.class.inc

```diff
@@ -60,24 +60,44 @@ class NDB_Menu_Filter_Imaging_Browser extends NDB_Menu_Filter
      */
     function _setupVariables()
     {
+        $config = NDB_Config::singleton();
+        $DB     = Database::singleton();
+        $toTable_scan_types = $config->getSetting('tblScanTypes');
+
+        $all_scan_types = $DB->pselect(
+            "SELECT ID, Scan_type FROM mri_scan_type mri",
+            array()
+        );
+        foreach ($all_scan_types as $row) {
+            $type = $row['Scan_type'];
+            $all_scan_types[$row['ID']] = $type;
+        }
+
+        $scan_id_types =array_flip(
```

Comment:
This looks like it's trying to be clever enough that it should have a comment explaining it.